tools/marvin/marvin/lib/base.py

```diff
@@ -3377,6 +3377,40 @@ def list(cls, apiclient, **kwargs):
             cmd.listall = True
         return(apiclient.listPortableIpRanges(cmd))
 
+
+class NuageUnderlayPublicIpRange:
```

Comment:
Agree, but this is a key infrastructure query API for Nuage VSP SDN platform (overlay vs underlay networking). Moreover, our custom base/utility class "nuageTestCase.py" also imports and uses this class. Thus, we have added it to the Marvin base class "base.py" for cleaner abstraction and code usage. 

For other Nuage VSP supported feature specific/custom APIs, we tend to include them in our custom base/utility class "nuageTestCase.py" at test/integration/plugins/nuagevsp/. 

Let me know, if you still think otherwise.